internal/service/resourcepolicy/resource.go

```diff
@@ -64,7 +52,7 @@ func (r *resourcePolicyRS) ModifyPlan(ctx context.Context, req resource.ModifyPl
 		return
 	}
 	sdkCreate := &admin.ApiAtlasResourcePolicyCreate{
-		Name:     randPolicyName(16),
+		Name:     *name,
```

Comment:
do we need to check pointer is not nil to avoid panics?

Reply:
it is checked in the lines above